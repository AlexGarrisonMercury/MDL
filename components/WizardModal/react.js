import React, { Component } from 'react'
import PropTypes from 'prop-types'
import onClickOutside from 'react-onclickoutside'
import classNames from 'classnames/bind'
import ModalHeader from './react/ModalHeader'
import ModalBody from './react/ModalBody'
import ModalFooter from './react/ModalFooter'
import style from './style.module.scss'

const cx = classNames.bind(style)

export class Modal extends Component {
  handleClickOutside = () => this.props.onOutsideClick()

  render() {
    const { children } = this.props
    return <div className={cx('wizard-modal')}>{children}</div>
  }
}

Modal.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
  onOutsideClick: PropTypes.func,
}

Modal.defaultProps = {
  onOutsideClick: undefined,
}

const WizardModal = onClickOutside(Modal)

export default WizardModal

WizardModal.Header = ModalHeader
WizardModal.Body = ModalBody
WizardModal.Footer = ModalFooter
