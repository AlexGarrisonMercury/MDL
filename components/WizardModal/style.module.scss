@import "styles/index";

.wizard-modal {
  position: relative;
  width: 372px;
  height: 610px;
  box-sizing: border-box;

  &__header {
    position: relative;
    width: 100%;
    margin: 0;
    border: none;
    background: $color-white;
    fill: $color-limerick;
    box-sizing: border-box;
  }

  &__header-title-area {
    display: flex;
    align-items: center;
    padding: $space-m;
    padding-top: 20px;
  }

  &__header-icon {
    @include square($space-m);
    display: inline-block;
    margin-right: $space-xs;
  }

  &__header-icon-svg {
    width: 100%;
    height: 100%;
  }

  &__field-group-label {
    margin-top: $space-xs;
    margin-bottom: $space-s;
  }

  &__close-icon {
    @include square($space-m);
    position: absolute;
    right: $space-m;
    fill: $color-deepgrey;
    cursor: pointer;
  }

  &__close-icon-svg {
    width: 100%;
    height: 100%;
  }

  &__subheading {
    margin: 0;
    padding: $space-s $space-m 0;
    font-weight: 300;
    text-transform: uppercase;
    background: $color-white;
  }

  &__title {
    color: $color-limerick;
    font-size: 24px;
    font-weight: 300;
    margin: 0;
    text-transform: uppercase;
  }

  &__body {
    height: 377px;
    background: $color-snow;
    overflow: hidden;
    overflow-y: scroll;
    padding: $space-m;

    &::-webkit-scrollbar {
      width: 6px;
      background: transparent;
    }

    &::-webkit-scrollbar-thumb {
      background: $color-silver;
    }
  }

  &__footer {
    display: flex;
    align-items: center;
    justify-content: space-between;
    background: $color-white;
    padding: $space-m;
  }
}
