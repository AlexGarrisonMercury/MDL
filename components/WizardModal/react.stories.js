import React from 'react'
import { storiesOf } from '@storybook/react'
import { mGlobe } from '@masonite/svg-icons'
import { React as ModalBackdrop } from 'components/ModalBackdrop'
import { React as PaginationDots } from 'components/PaginationDots'
import { React as SmallTitle } from 'components/SmallTitle'
import { React as TextField } from 'components/TextField'
import { React as Button } from 'components/Button'
import Modal from './react'

storiesOf('WizardModal', module).add('example', () => (
  <ModalBackdrop>
    <Modal>
      <Modal.Header
        title="Hello World"
        subtitle="Subtitle or Subheading"
        icon={mGlobe}
        onDismiss={() => { }}
      />
      <Modal.Body>
        <SmallTitle>Small Title as Field Label</SmallTitle>
        <TextField placeholder="Some Field" onChange={() => { }} value="" variant="light" />
      </Modal.Body>
      <Modal.Footer activePage={0} totalPages={4}>
        <Button variant="secondary" uppercase>
          Save
        </Button>
      </Modal.Footer>
    </Modal>
  </ModalBackdrop>
))
