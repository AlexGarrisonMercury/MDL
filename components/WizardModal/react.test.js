import React from 'react'
import { mount } from 'enzyme'
import WizardModal from './react'

describe('WizardModal', () => {
  it('should render without crashing', () => {
    expect(
      mount(
        <WizardModal>
          <WizardModal.Header title="Title" />
          <WizardModal.Body>Body</WizardModal.Body>
          <WizardModal.Footer>Footer</WizardModal.Footer>
        </WizardModal>,
      ),
    ).toMatchSnapshot()
  })
})
