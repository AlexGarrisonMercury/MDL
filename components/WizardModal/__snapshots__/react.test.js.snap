// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardModal should render without crashing 1`] = `
<OnClickOutside(Modal)
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
>
  <Modal
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
  >
    <div
      className="wizard-modal"
    >
      <ModalHeader
        title="Title"
      >
        <header
          className="wizard-modal__header"
        >
          <div
            className="wizard-modal__header-title-area"
          >
            <div
              className="wizard-modal__header-icon"
            >
              <svg
                className="wizard-modal__header-icon-svg"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
              />
            </div>
            <h1
              className="wizard-modal__title"
            >
              Title
            </h1>
          </div>
        </header>
      </ModalHeader>
      <ModalBody>
        <div
          className="wizard-modal__body"
        >
          Body
        </div>
      </ModalBody>
      <ModalFooter>
        <footer
          className="wizard-modal__footer"
        >
          Footer
        </footer>
      </ModalFooter>
    </div>
  </Modal>
</OnClickOutside(Modal)>
`;
